lib/install/management/delete.go

```diff
@@ -230,11 +241,44 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *config.Vir
 
 	var err error
 	var children []*vm.VirtualMachine
-	if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
-		return err
+
+	vchFolder, err := d.appliance.Folder(d.op)
+	if err != nil {
+		d.op.Errorf("Failed to obtain the VCH folder: %s", err)
+		return fmt.Errorf("Failed to obtain the VCH folder. See vic-machine.log for more details. ")
+	}
+	// set the VCH Folder on session
+	d.session.VCHFolder = vchFolder
+	// if we have a vchFolder and it's not the VMFolder then gather the children via the folder
+	if vchFolder != nil && vchFolder.Reference() != d.session.VMFolder.Reference() {
+		// vch parent inventory folder exists, get the children from it.
+		d.op.Debugf("Finding children in VCH Folder: %s", vchFolder.InventoryPath)
+		folderChildren, err := vchFolder.Children(d.op)
+		if err != nil {
+			return err
+		}
+
+		for _, child := range folderChildren {
```

Comment:
same code as in `portlayer/exec/container.go` - should we have a Folder.ChildVMs method instead?

Reply:
agreed..we should...I mention that somewhere...possibly in my upgrade PR..